File: 프로그래머스/2/49993. 스킬트리/스킬트리.java
import java.util.*;

class Solution {
    public int solution(String skill, String[] skill_trees) {
        
        ArrayList<Character> turn = new ArrayList<>();
        
        for(int i=0; i<skill.length(); i++){
            turn.add(skill.charAt(i));
        }
            
        int answer = 0;
        
        for(int i=0; i<skill_trees.length; i++){
            int index = 0;
            String now = skill_trees[i];
            boolean possible = true;
            for(int j=0; j<now.length(); j++){
                if(turn.contains(now.charAt(j))){
                    if(turn.get(index) == (now.charAt(j))){
                        index++;
                    }else{
                        possible = false;
                        break;
                    }
                }
            }
            if(possible) answer++;
        }
        
        return answer;
    }
}

File: 프로그래머스/2/49993. 스킬트리/스킬트리.js
function solution(skill, skill_trees) {
    
    const turn = [];
    
    for(let i=0; i<skill.length; i++){
        turn.push(skill[i]);
    }
    
    var answer = 0;

    for(let i=0; i<skill_trees.length; i++){
        var index = 0;
        var now = skill_trees[i];
        var possible = true;
        for(let j=0; j<now.length; j++){
            if(turn.indexOf(now.charAt(j)) >= 0){
                if(turn[index] == now.charAt(j)){
                    index++;
                }else{
                    possible = false;
                    break;
                }
            }
        }
        if(possible) answer++
    }
    
    return answer;
}